adagrams/game.py

```diff
@@ -1,11 +1,109 @@
+import random
+
+LETTER_DISTRIBUTION = {
+    "A": 9,
+    "B": 2,
+    "C": 2,
+    "D": 4,
+    "E": 12,
+    "F": 2,
+    "G": 3,
+    "H": 2,
+    "I": 9,
+    "J": 1,
+    "K": 1,
+    "L": 4,
+    "M": 2,
+    "N": 6,
+    "O": 8,
+    "P": 2,
+    "Q": 1,
+    "R": 6,
+    "S": 4,
+    "T": 6,
+    "U": 4,
+    "V": 2,
+    "W": 2,
+    "X": 1,
+    "Y": 2,
+    "Z": 1
+}
+
+SCORE_CHART = {
+    1: ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],  
+    2: ["D", "G"],
+    3: ["B", "C", "M", "P"],
+    4: ["F", "H", "V", "W", "Y"],
+    5: ["K"],
+    8: ["J", "X"],
+    10: ["Q", "Z"]
+}
+
 def draw_letters():
-    pass
+    letter_pool = []
+    letter_bank = []
+    for letter, count in LETTER_DISTRIBUTION.items():
+        for i in range(count):
+            letter_pool.append(letter)
+    # Should we rewrite 45-47 with list comprehension? This is what we tried:
+    # letter_pool = [[letter] * count
+    #               for letter, count in LETTER_DISTRIBUTION.items()]
+    for letter in range(10):
+        random_letter = random.choice(letter_pool)
+        letter_bank.append(random_letter)
+        letter_pool.remove(random_letter)
+    return letter_bank
 
 def uses_available_letters(word, letter_bank):
-    pass
+    bank_dict = {}
+    # We tried to implement dictionary comprehension for this,
+    # but are unsure if it's possible:
+    # alt_bank_dict = {letter, 1 for letter in letter_bank
+    #                  if item not in alt_bank_dict else letter, + 1}
+    for item in letter_bank:
+        if item not in bank_dict:
+            bank_dict[item] = 1
+        else:
+            bank_dict[item] += 1
+    for letter in word.upper():
+        if letter not in bank_dict.keys():
+            return False
+        elif letter in bank_dict.keys():
+            bank_dict[letter] -= 1
+        
+        if bank_dict[letter] < 0:
+            return False
+    return True
 
 def score_word(word):
-    pass
+    total_score = 0
+    for character in word.upper():
+        for tally, letters in SCORE_CHART.items():
+            if character in letters:
+                total_score += tally
+    if len(word) >= 7:
+        total_score += 8
+    return total_score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    word_info = [{"word": word, "score": score_word(word), "length": len(word)}
```

Comment:
This function works great! One minor improvement would be to avoid looping by checking whether there's only one element in `word_info`. Since that list only contains more than one element if there's ties, you can use that fact to return the sole element, e.g.
```python
    if len(word_info) == 1:
        return word_info[0]["word"], word_info[0]["score"]
```
Then go on to do all the tie-break looping only if you actually have a tie.